azkaban-web-server/src/main/java/azkaban/webapp/AzkabanWebServer.java

```diff
@@ -228,7 +230,10 @@ public static void launch(final AzkabanWebServer webServer) throws Exception {
     /* This creates the Web Server instance */
     app = webServer;
 
-    webServer.executorManager.start();
+    // Todo jamiesjc: also start the threads in ExecutionController if needed.
+    if (webServer.executorManagerAdapter instanceof ExecutorManager) {
```

Comment:
Often `instanceof` check is considered anti-pattern. What if we had yet another, third implementation of `ExecutorManagerAdapter` that is not ExecutorManager but behaves more like it than like the new ExecutionController?
Could we delegate and trust the implementation to do the right thing? In this case, could we call `start` unconditionally here, but make `start` implementation a noop in the new class?

Reply:
Yes, as I mentioned in the "todo", after we add the implementation for `start` in the new class, we can remove this `instanceof` check. This would also require adding the `start` in the `ExecutorManagerAdapter` interface.